src/main/java/uk/gov/companieshouse/api/accounts/service/impl/ProfitAndLossService.java

```diff
@@ -142,15 +147,17 @@ public ProfitAndLossService(
     }
 
     @Override
-    public ResponseObject<ProfitAndLoss> delete(String companyAccountsId, HttpServletRequest request)
+    public ResponseObject<ProfitAndLoss> delete(String companyAccountsId, HttpServletRequest request, AccountingPeriod period )
         throws DataException {
 
-        String profitLossId = generateID(companyAccountsId, request);
+        String profitLossId = generateID(companyAccountsId, period);
+
+        invalidateStatementsIfExisting(companyAccountsId, request);
```

Comment:
Should this be further down? Perhaps just before `return new ResponseObject<>(ResponseStatus.UPDATED);` so we only invalidate statements if the db delete was successful?